test/download_helper.js

```diff
@@ -0,0 +1,257 @@
+    });
+
+    it('button should have specific properties', async () => {
+      dh.button = document.createElement("button");
+
+      dh.enableButton();
+
+      expect(dh.button.disabled).to.be.false;
+      expect(dh.button.textContent == "Download curl File").to.be.true;
+    });
+  });
+
+  describe("disableButton()", () => {
+    it('button listener should be removed', async () => {
+      dh.button = document.createElement("button")
+      dh.button.removeEventListener = sinon.spy();
+
+      dh.disableButton();
+
+      expect(dh.button.removeEventListener.calledOnce).to.be.true;
+
+    });
+    it('button should have specific properties', async () => {
+      dh.button = document.createElement("button")
+
+      dh.disableButton();
+
+      expect(dh.button.disabled).to.be.true;
+      expect(dh.button.textContent == "preparing download").to.be.true;
+    });
+  });
+
+  describe("callback()", () => {
+    it('enables button when all style.display blank', async () => {
+      createDomNodes(`
+        <div id="testId" class="download-title"><div class="item-button" style="display: none"></div></div>
+        <div id="testId" class="download-title"><div class="item-button" style="display: none"></div></div>
+        <div id="testId" class="download-title"><div class="item-button" style="display: none"></div></div>
+      `)
+
+      dh.enableButton = sinon.spy();
+
+      dh.mutationCallback();
+      expect(dh.enableButton.notCalled).to.be.true;
+
+      //change DOM nodes
+      let nodes = document.querySelectorAll(".download-title .item-button")
+      nodes.forEach(function(element, index) {
+        element.style.display = "";
+      });
+      
+      dh.mutationCallback();
+      expect(dh.enableButton.calledOnce).to.be.true;
+    });
+
+    it('disables button when all style.display ""', async () => {
+      createDomNodes(`
+        <div id="testId" class="download-title"><div class="item-button" style="display:"></div></div>
+        <div id="testId" class="download-title"><div class="item-button" style="display:"></div></div>
+        <div id="testId" class="download-title"><div class="item-button" style="display:"></div></div>
+      `)
+
+      dh.disableButton = sinon.spy();
+
+      dh.mutationCallback();
+      expect(dh.disableButton.notCalled).to.be.true;
+
+      //change DOM nodes
+      let nodes = document.querySelectorAll(".download-title .item-button")
+      nodes.forEach(function(element, index) {
+        element.style.display = "none";
+      });
+
+      dh.mutationCallback();
+      expect(dh.disableButton.calledOnce).to.be.true;
+    });
+  });
+
+  describe("init()", () => {
+    it('creates button', async () => {
+      dh.createButton = sinon.spy();
+
+      dh.init();
+
+      expect(dh.createButton.called).to.be.true;
+    });
+
+    it('calls mutationCallback', async () => {
+      dh.mutationCallback = sinon.spy();
+
+      dh.init();
+
+      expect(dh.mutationCallback.called).to.be.true;
```

Comment:
Similar to the previous comment about Chai message types: [sinon-chai](https://www.chaijs.com/plugins/sinon-chai/)

`expect(dh.mutationCallback).to.have.been.called;`